api_tests/src/actors/balance_asserter.ts

```diff
@@ -0,0 +1,194 @@
+import { EthereumWallet } from "../wallets/ethereum";
+import { BitcoinWallet } from "../wallets/bitcoin";
+import { LightningWallet } from "../wallets/lightning";
+
+export interface BalanceAsserter {
+    assertSwapped(): Promise<void>;
+    assertRefunded(): Promise<void>;
+    assertNothingReceived(): Promise<void>;
+}
+
+export class Erc20BalanceAsserter implements BalanceAsserter {
+    public static async forFunder(
+        wallet: EthereumWallet,
+        swapAmount: bigint,
+        tokenContract: string
+    ) {
+        await wallet.mintErc20(swapAmount, tokenContract); // As funder, we need money to spend
+
+        const startingBalance = await wallet.getErc20Balance(tokenContract, 0);
+        const expectedDiff = swapAmount * BigInt(-1); // As funder, we expect our balance to decrease
+
+        return new Erc20BalanceAsserter(
+            wallet,
+            startingBalance,
+            expectedDiff,
+            tokenContract
+        );
+    }
+
+    public static async forRedeemer(
+        wallet: EthereumWallet,
+        swapAmount: bigint,
+        tokenContract: string
+    ) {
+        const startingBalance = await wallet.getErc20Balance(tokenContract, 0);
+
+        return new Erc20BalanceAsserter(
+            wallet,
+            startingBalance,
+            swapAmount,
+            tokenContract
+        );
+    }
+
+    constructor(
+        private readonly wallet: EthereumWallet,
+        private readonly startingBalance: bigint,
+        private readonly expectedDiff: bigint,
+        private readonly tokenContract: string
+    ) {}
+
+    public async assertSwapped() {
+        const currentBalance = await this.wallet.getErc20Balance(
+            this.tokenContract,
+            0
+        );
+        const expectedBalance = this.startingBalance + this.expectedDiff;
+
```

Comment:
As I read this I wondered, because of the name, if this checked the balance of both wallets i.e., that funds have gone from one and arrived in the other (for only this actor). (Obviously this is impossible since we only have a handle to one wallet.) Since this is not the case and we only check one wallet, would `assertRedeemed` not be a better name?

Reply:
Good point, I will change the function name :)